data/transactions/verify/txn.go

```diff
@@ -346,12 +353,21 @@ func logicSigBatchVerify(txn *transactions.SignedTxn, groupIndex int, groupCtx *
 	if groupIndex < 0 {
 		return errors.New("Negative groupIndex")
 	}
+
+	var ts int64
+	if groupCtx.consensusParams.EnableFirstValidTimeStamp {
+		ts, err = groupCtx.timestamper.BlockTimeStamp(txn.Txn.FirstValid.SubSaturate(1))
+		if err != nil {
+			return err
+		}
```

Comment:
I don't think it should subsaturate, it should error if FirstValid = 0.  You are asking for the timestamp of block -1, which doesn't exist.  Perhaps it could return the timestamp in the genesis file, but that seems too cute.

Reply:
BlockTimeStamp does not err anymore, deferring resolution to logic.Eval